src/core/js/scrolling.js

```diff
@@ -0,0 +1,139 @@
+                return selectorScrollTo.apply($app, arguments);
+            };
+        },
+
+        _fixBrowser: function() {
+            var app = Adapt.scrolling.$app[0];
+            window.scrollTo = function(x, y) {
+                app.scrollTop = y || 0;
+                app.scrollLeft = x || 0;
+            };
+            var $window = $(window);
+            this.$app.on("scroll", function() {
+                $window.scroll();
+            });
+        }
+
+    });
+
+    Adapt.scrolling = new Scrolling();
+
+    Adapt.scrollTo = function(selector, settings) {
+        // Get the current location - this is set in the router
+        var location = (Adapt.location._contentType) ?
+            Adapt.location._contentType : Adapt.location._currentLocation;
+        // Trigger initial scrollTo event
+        Adapt.trigger(location+':scrollTo', selector);
+        //Setup duration variable passed upon arguments
+        var settings = (settings || {});
+        var disableScrollToAnimation = Adapt.config.has('_disableAnimation') ? Adapt.config.get('_disableAnimation') : false;
+        if (disableScrollToAnimation) {
+            settings.duration = 0;
+        }
+        else if (!settings.duration) {
+            settings.duration = $.scrollTo.defaults.duration;
+        }
+
+        var offsetTop = 0;
+        if (Adapt.scrolling.isLegacyScrolling) {
+            offsetTop = -$(".navigation").outerHeight();
+            // prevent scroll issue when component description aria-label coincident with top of component
+            if ($(selector).hasClass('component')) {
+                offsetTop -= $(selector).find('.aria-label').height() || 0;
+            }
+        }
+
+        if (!settings.offset) settings.offset = { top: offsetTop, left: 0 };
+        if (settings.offset.top === undefined) settings.offset.top = offsetTop;
+        if (settings.offset.left === undefined) settings.offset.left = 0;
+
+        if (settings.offset.left === 0) settings.axis = "y";
+
+        if (Adapt.get("_canScroll") !== false) {
+            // Trigger scrollTo plugin
+            $.scrollTo(selector, settings);
+        }
+
+        // Trigger an event after animation
+        // 300 milliseconds added to make sure queue has finished
+        _.delay(function() {
+            $(selector).a11y_focus();
+            Adapt.trigger(location+':scrolledTo', selector);
+        }, settings.duration+300);
+    };
+
+    Adapt.getOffset = function(selector) {
+        var $app = Adapt.scrolling.$app;
+        var $element = $(selector);
+        var offset = $element.offset();
+        var $stack = $element.parents().add($element);
+        var $scrollParents = $stack.filter('#app');
+        $scrollParents.each(function(index, parent) {
+            var $parent = $(parent);
+            var scrolltop = parseInt($parent.scrollTop());
+            var scrollleft = parseInt($parent.scrollLeft());
+            offset.top += scrolltop - $app.offset()['top'];
```

Comment:
I can understand adding the scrollTop/scrollLeft values so you can discount the effects of scrolling to get the offset position, but I'm, confused by the subtraction of `$app.offset()['top']`. Let's say the $app was offset from the top by 50px, then if $element is 10 levels deep under the document you'll subtract 10 x 50px (well 9 x 50px) - I must be interpreting this badly - can you help?

Reply:
There will only ever bee one `$scrollParents` the `#app` element. I should rewrite this to make that clearer.

Line 127: Make array of parents plus subject element
Line 128: Filter array so that only `#app` element persists
Line 129: Iterate through array.

I'll rewrite it.

Reply:
Oh that was a bad read of `filter`, sorry! So, with there only being one `#app`, why bother with line 128 and the loop?

Reply:
The code evolved over a few iterations. It just ended up like that.